torch_ipex/csrc/auto_opt_config.h

```diff
@@ -25,15 +29,134 @@ class AutoOptConfig {
     return jit_fuse_;
   }
 
+  // bf16
   inline void set_mix_bf16_fp32(bool value) {
     mix_bf16_fp32_ = value;
   }
   inline bool get_mix_bf16_fp32() {
     return mix_bf16_fp32_;
   }
 
+  // int8
+  inline void set_mix_int8_fp32(bool value) {
+    mix_int8_fp32_ = value;
+  }
+  inline bool get_mix_int8_fp32() {
+    return mix_int8_fp32_;
+  }
+
+  inline void set_int8_calibration(bool value) {
+    calibration_step_ = value;
+  }
+  inline bool get_int8_calibration() {
+    return calibration_step_;
+  }
+
+  inline void insert_or_updata_observer(std::string op_name, std::vector<float> max_values) {
+    num_ops_id++;
+    if (observers_.size() < num_ops_id) {
+      //Operator op = {num_ops_id - 1, op_n};
+      Observer new_observer = {num_ops_id - 1, op_name, max_values};
+      observers_.push_back(new_observer);
+    } else {
+      for (auto i = 0; i < max_values.size(); i++)
+        observers_[num_ops_id -1].max_values[i] = std::max(observers_[num_ops_id -1].max_values[i], max_values[i]);
+    }
+  }
+
+  inline void print_observer() {
+    for (auto i = 0; i< observers_.size(); i++) {
+      for (auto j = 0; j < observers_[i].max_values.size(); j++)
+        std::cout<<observers_[i].max_values[j]<<std::endl;
+    }
+  }
+
+  inline void print_indicator() {
+    for (auto i = 0; i< indicators_.size(); i++) {
+      auto scales = indicators_[i].get_indicator_scales();
+      for (auto j = 0; j< scales.size(); j++)
+          std::cout<<scales[j]<<std::endl;
+    }
+  }
+
+  inline void add_indicators() {
+    num_ops_id = 0;
+    // default used is s8
+    for (auto i = 0; i < observers_.size(); i++) {
+      std::vector<float> scales;
+      std::vector<bool> uint8_used;
+      for (auto j = 0; j < observers_[i].max_values.size(); j++) {
+        scales.push_back(127.5 / observers_[i].max_values[j]);
+        uint8_used.push_back(false);
+      }
+        // zero_points not used now, zero_points = 0 for u8 and 128 for s8.
+        //zero_point = 128;
+      Indicator new_indicator(observers_[i].Id, observers_[i].Name, scales, uint8_used, true);
+      indicators_.push_back(new_indicator);
+    }
+    observers_.clear();
+  }
+
+  inline std::tuple<std::vector<float>, bool> get_indicator_scales(std::vector<bool> uint8_used) {
```

Comment:
How do we decide what methods are defined in the header file and what are in the source file? Having too many big functions in the header files doesn't sound a good practice, e.g. would slow down the incremental build etc.